File: CS50's Introduction to Computer Science/pset6/caesar/caesar.py
# program that encrypts messages using Caesar’s cipher
import sys

# program needs to accept a command-line argument, key
if len(sys.argv) != 2 or not sys.argv[1].isdigit():
    print("Usage: python caesar.py key")
    sys.exit(1)

key = int(sys.argv[1]) % 26
plaintext = input("plaintext: ")
ciphertext = ""

# Caesar "encrypts" confidential messages by shifting each letter therein by some number of places.
# For instance (if key = 1), A as B, B as C, C as D, …, and, wrapping around alphabetically, Z as A.
for c in plaintext:
    if c.isalpha():
        if c.islower():
            ciphertext += chr((ord(c) - ord("a") + key) % 26 + ord("a"))
        if c.isupper():
            ciphertext += chr((ord(c) - ord("A") + key) % 26 + ord("A"))
    else:
        ciphertext += c
print("ciphertext: ", ciphertext)

File: CS50's Introduction to Computer Science/pset2/caesar/caesar.c
// Program that encrypts message using Caesar’s cipher
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{
    // Check for a single command-line argument and a non-negative integer
    if (argc == 2 && atoi(argv[1]) >= 0)
    {
        string plaintext = get_string("plaintext: ");
        string ciphertext = plaintext;

        // Loop throught the plaintext string character by character
        for (int i = 0; i < strlen(plaintext); i++)
        {
            // Check if character is an alphabetic letter
            if (isalpha(plaintext[i]))
            {
                // Check if character is lowercase
                if (islower(plaintext[i]))
                {
                    // For lowercase: ASCII values ('a' is 97) to alphabetical indexes ("a" is 0).
                    // Caesar’s algorithm (i.e., cipher) encrypts messages by "rotating" each letter by k positions.
                    // c[i]=(p[i]+k)mod26
                    // atoi - convert a string to an integer
                    // Alphabetical indexes to ASCII values
                    ciphertext[i] = ((plaintext[i] - 97 + atoi(argv[1])) % 26 + 97);
                }
                else
                {
                    // For uppercase: ASCII values ('A' is 65) to alphabetical indexes ("A" is 0).
                    // Caesar's algorithm
                    // Alphabetical indexes to ASCII values
                    ciphertext[i] = ((plaintext[i] - 65 + atoi(argv[1])) % 26 + 65);
                }
            }
            else
            {
                // If character is not letter
                ciphertext[i] = plaintext[i];
            }
        }

        // Print of ciphertext and return 0
        printf("ciphertext: %s\n", ciphertext);
        return 0;
    }
    else
    {
        // If command-line check is not passed we print error message and return 1
        printf("Error: program accepts a single command-line argument, a non-negative integer.\n");
        return 1;
    }
}